### DISCUSSION TITLE
Dynamic name to trigger data updates

### DISCUSSION THREAD

**Discussion by ercanertan** [alpinejs/alpine#1795] [1. Help]
```
<div
    x-data="{ title: 'Hello' }"
    @set-title.window="title = $event.detail"
 >
    <h1 x-text="title"></h1>
 </div>

<div x-data>
    <button @click="$dispatch('set-title', 'Hello World!')">...</button>
</div>
```

What I want to achieve is give dynamic name to trigger data updates. 

@set-title+id.window (-> How can make this dynamic ?)
$dispatch('set-title'+id, 'Hello World!')  (This is fine)

Thanks in advance

**Comment by danddanddand** [comment]
I'm not sure I understand your question or intent but the event handler name isn't interpreted as javascript, just a string parsed into directive names and modifiers so this wouldn't be possible with current way directives are created. 

You could implement directive that used modifier (like the .window is a modifier) for the value you wanted to append and implement it but I don't think the use cases would be common enough you'd want to. 

You probably want more sophisticated event handler strategy where you don't rely on the event name + id on window to make the handler unique. You could always dispatch more info using an object and handle it that way. for example $dispatch('set-title', {title:'Hello World!',id:id}). 

Hope this gives you some ideas. If you want to give more details on your intent might help me understand what you're trying to achieve.

**Reply by ercanertan** [reply]
Thank you for your reply @danddanddand ,

The reason I need this because I am trying to create reusable file selector from modal. 

1. Click "Browse" button
2. Open Modal and show files
3. Select file and close the modal
4. Get the selected file information back and use it on related input.

Apply same thing to another input with different file...

After long digging , I was thinking same approach like you suggested 
$dispatch('set-title', {title:'Hello World!',id:id})
